### ISSUE TITLE
🐞 [BUG] "npm ERR! cb() never called!" or "Exit handler never called"

### ISSUE THREAD

**Issue by darcyclarke** [npm/cli#417]
### \*Updated\* as of 01/15/2021 

**Note: Please read [this doc](https://github.com/npm/cli/wiki/%22cb()-never-called%3F-Exit-handler-never-called%3F-I'm-having-the-same-problem!%22) before filing a new issue.**

**Comment by DanielRuf** [comment]
I think we have many reports of the same so these should all be duplicates.

https://github.com/npm/cli/issues/423
https://github.com/npm/cli/issues/425
https://github.com/npm/cli/issues/442
https://github.com/npm/cli/issues/451
https://github.com/npm/cli/issues/455
https://github.com/npm/cli/issues/465

Afaik this is resolved by force-clearing the cache and upgrading / updating to the latest npm version.

**Comment by DanielRuf** [comment]
https://github.com/npm/cli/issues/303
https://github.com/npm/cli/issues/306
https://github.com/npm/cli/issues/325
https://github.com/npm/cli/issues/353
https://github.com/npm/cli/issues/369
https://github.com/npm/cli/issues/370
https://github.com/npm/cli/issues/375
https://github.com/npm/cli/issues/383
https://github.com/npm/cli/issues/408
https://github.com/npm/cli/issues/418
https://github.com/npm/cli/issues/448
https://github.com/npm/cli/issues/474
https://github.com/npm/cli/issues/499
https://github.com/npm/cli/issues/522

**Comment by ruyadorno** [comment]
thanks @DanielRuf for linking all those ❤️

**Comment by ruyadorno** [comment]
#489

**Comment by DanielRuf** [comment]
https://github.com/npm/cli/issues/544
https://github.com/npm/cli/issues/552
https://github.com/npm/cli/issues/553
https://github.com/npm/cli/issues/556
https://github.com/npm/cli/issues/566
https://github.com/npm/cli/issues/570
https://github.com/npm/cli/issues/571
https://github.com/npm/cli/issues/573
https://github.com/npm/cli/issues/574
https://github.com/npm/cli/issues/581
https://github.com/npm/cli/issues/584
https://github.com/npm/cli/issues/585
https://github.com/npm/cli/issues/594
https://github.com/npm/cli/issues/596
https://github.com/npm/cli/issues/618
https://github.com/npm/cli/issues/630
https://github.com/npm/cli/issues/634

**Comment by DanielRuf** [comment]
https://github.com/npm/cli/issues/573

**Comment by DanielRuf** [comment]
https://github.com/npm/cli/issues/574

**Comment by DanielRuf** [comment]
https://github.com/npm/cli/issues/581

**Comment by DanielRuf** [comment]
I guess we can check which version first introduced this (doing a small `git bisect`) and check the stacktraces for similarities.

**Comment by DanielRuf** [comment]
Following versions were mentioned in the issues:

6.4.1
6.9.0
6.10.2
6.12.1
6.13.1
6.13.4

Now trying to get a reproducible testcase.

**Comment by DanielRuf** [comment]
Relevant changes in the past tolog this error: https://github.com/npm/npm/pull/15716

**Comment by DanielRuf** [comment]
Tests with a local npm 6.13.1 (on macOS):

`@vue/cli`: not reproducible
`npm audit fix`: not reproducible
`plotly.js`: not reproducible
`expo-cli`: not reproducible

So far it looks like this is caused by other errors which cancel the CLI too early.

I remember that we had this bug also on Ubuntu with the latest version and some packages.

**Comment by melvinwallerjr** [comment]
I'm experiencing the same error trying to install a company project for development.
Microsoft Windows [Version 10.0.17134.1184]

Using NVM to switch instances of node/npm
node v10.14.2 (64-bit), npm v6.4.1
node v12.4.0 (64-bit), npm v6.9.0

project was generated with Angular CLI version 8.3.21

[NVM for Windows Setup, Ecor Ventures LLC, ‎Tuesday, ‎August ‎7, ‎2018 9:46:31 PM]

**Comment by ljharb** [comment]
(note that must be nvm-windows; actual nvm doesn't work on non-WSL Windows, and doesn't distinguish 64-bit)

**Comment by mdouglass** [comment]
We're getting this error intermittently in both local development and in our CI system. Happy to add any additional logging that might help. Attached the npm log from a run that just happened on my laptop on my mac -- trying to npm install one of our private packages. Immediately re-running the command worked without issue.

`npm install @globalworldwide/km-core@latest`
[2020-01-04T02_02_56_202Z-debug.log](https://github.com/npm/cli/files/4021438/2020-01-04T02_02_56_202Z-debug.log)

❯ npm -v
6.13.4
❯ node -v
v13.5.0

Let me know if there's anything I can do to help track this down, would like to kill the issue.

**Comment by DanielRuf** [comment]
https://github.com/npm/cli/issues/682

**Comment by DanielRuf** [comment]
https://github.com/npm/cli/issues/671

**Comment by DanielRuf** [comment]
https://github.com/npm/cli/issues/665

**Comment by mikemimik** [comment]
Actions Triggered:
- `npm audit fix` (`npm@6.9.0` / `node@10.16.1`)
- `npm install` (`npm@6.10.2` / `node@12.8.0`)
- `sudo npm install -g npm` (`npm@6.9.0` / `node@x.y.z`)
- n/a _likely_ `npm i` (`npm@6.12.1` / `node@13.1.0`) (`npm@6.12.0` / `node@12.13.0`)
- `npm install` (`npm@6.4.1` / `node@12.13.0`)
- `npm install` (`npm@6.9.0` / `node@10.16.1`)
- `vue create <app>` (`npm@6.2.0` / `node@10.9.0`)

Thoughts as triaging:
- Seems like there is an error while installing a package that causes this `cb() never called!` error.
- There are 42 issues to triage...
- https://github.com/npm/cli/issues/442 original issue references a fix which references updating and cleaning cache to solve issue (updating fixes permissions), cache might indicate problem exists/existed in `cacache` or `pacote`
- https://github.com/npm/cli/issues/451 seems to have same symptom but cause doesn't seem to be within `npm`, though it does help point to an issue with while installing a package

**Comment by puppy0cam** [comment]
It's shocking how few people know how to use a search bar...

**Comment by jslegers** [comment]
Does anyone have ANY idea what might be causing this?

Are there any workarounds?

When I run my `npm install` locally, on windows 10, everything works fine.

When I run my `npm install` on dev.azure.com, on a 64bit Amazon Linux/4.13.0 machine, I get the following error :

````
120982 error cb() never called!
120983 error This is an error with npm itself. Please report this error at:
120984 error <https://npm.community>
````

I tried upgrading my Node environment from Node 10 to Node 12, as I was also running Node 12 locally, but this did not seem to have any impact whatsoever.

I have no idea what to do with this, and this issue is totally blocking!!

**Comment by mikemimik** [comment]
Hey @jslegers sorry to hear you're blocked! 

In my [initial research](https://github.com/npm/cli/issues/417#issuecomment-575834563) into the issue, I noted that the issue https://github.com/npm/cli/issues/442 referenced our community page which had a link to a possible solution for you. Clearing the cache seemed to solve the issue for some. I would suggest giving that a try to see if you can unblock your work.

**Comment by jslegers** [comment]
>Hey @jslegers sorry to hear you're blocked!
>
> In my initial research into the issue, I noted that the issue #442 referenced our community page which had a link to a possible solution for you. Clearing the cache seemed to solve the issue for some. I would suggest giving that a try to see if you can unblock your work.

Thanks for the tip!

I did eventually manage to find a solution on my own. 

Apparently, the issue was caused by my making some changes to the local package structure. As part of an ongoing refactoring attempt, I got rid of a one-size-fits-all kind of unscoped package and replaced it with a bunch of small scoped packages. Apparently, this confused NPM and caused the `npm ERR! cb() never called!` bug.

In seems to have been fixed by deleting my `package-lock.json` file and pushing that deletion to the remote branch where this issue occurred.

**Comment by jrtitus** [comment]
For me, this occurred when I was attempting to `npm install` over a corporate proxy, through a VPN (actually, the proxy doesn't seem to matter). NPM chokes when a package attempts to download a binary (for example, https://github.com/sass/node-sass/releases/download/v4.13.1/win32-x64-79_binding.node, in my case) during the postinstall step (no idea if it's package dependent).

After turning off the VPN (Global Protect) ~~and the proxy~~ - everything worked without a hitch.

# With Proxy and VPN turned ON ❌
```
$ npm i node-sass
npm ERR! cb() never called!

npm ERR! This is an error with npm itself. Please report this error at:
npm ERR!     <https://npm.community>

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\---\AppData\Roaming\npm-cache\_logs\2020-03-16T23_37_35_801Z-debug.log
```
![image](https://user-images.githubusercontent.com/1756049/76808390-0bf0d080-67be-11ea-8fca-5dbe6d0a005f.png)

![image](https://user-images.githubusercontent.com/1756049/76808475-49555e00-67be-11ea-925c-63fc51c91b77.png)

![image](https://user-images.githubusercontent.com/1756049/76810539-ddc2bf00-67c4-11ea-9c1a-3f73829d9959.png)

[2020-03-16T23_37_35_801Z-debug.log](https://github.com/npm/cli/files/4340808/2020-03-16T23_37_35_801Z-debug.log)

# With Proxy & VPN turned OFF ✔
```
$ npm i node-sass
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142

> node-sass@4.13.1 install C:\Users\---\Desktop\foo\node_modules\node-sass
> node scripts/install.js

Downloading binary from https://github.com/sass/node-sass/releases/download/v4.13.1/win32-x64-79_binding.node
Download complete
Binary saved to C:\Users\---\Desktop\foo\node_modules\node-sass\vendor\win32-x64-79\binding.node
Caching binary to C:\Users\---\AppData\Roaming\npm-cache\node-sass\4.13.1\win32-x64-79_binding.node

> node-sass@4.13.1 postinstall C:\Users\---\Desktop\foo\node_modules\node-sass
> node scripts/build.js

Binary found at C:\Users\---\Desktop\foo\node_modules\node-sass\vendor\win32-x64-79\binding.node
Testing binary
Binary is fine
npm WARN foo@1.0.0 No description
npm WARN foo@1.0.0 No repository field.

+ node-sass@4.13.1
added 173 packages from 137 contributors and audited 528 packages in 16.034s

3 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
```
![image](https://user-images.githubusercontent.com/1756049/76809415-21b3c500-67c1-11ea-801b-032b83b15a61.png)

# With Proxy Only ✔
```
$ npm i node-sass
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142

> node-sass@4.13.1 install C:\Users\---\Desktop\foo\node_modules\node-sass
> node scripts/install.js

Downloading binary from https://github.com/sass/node-sass/releases/download/v4.13.1/win32-x64-79_binding.node
Download complete
Binary saved to C:\Users\---\Desktop\foo\node_modules\node-sass\vendor\win32-x64-79\binding.node
Caching binary to C:\Users\---\AppData\Roaming\npm-cache\node-sass\4.13.1\win32-x64-79_binding.node

> node-sass@4.13.1 postinstall C:\Users\---\Desktop\foo\node_modules\node-sass
> node scripts/build.js

Binary found at C:\Users\---\Desktop\foo\node_modules\node-sass\vendor\win32-x64-79\binding.node
Testing binary
Binary is fine
npm WARN foo@1.0.0 No description
npm WARN foo@1.0.0 No repository field.

+ node-sass@4.13.1
added 173 packages from 137 contributors and audited 528 packages in 16.233s

3 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
```
![image](https://user-images.githubusercontent.com/1756049/76810553-e74c2700-67c4-11ea-8e8c-5006f62f39f0.png)

So, by deductive reasoning, it must have *something* to do with how the request/response is being handled over the VPN...

I made sure to `rm /c/Users/---/AppData/Roaming/npm-cache/node-sass/4.13.1/win32-x64-79_binding.node` and start with a new `node_modules` directory between each test.

I'm generally working from the office, so I only need to deal with the proxy, which doesn't cause an issue. However, now that we're all working from home, I had the chance to run into this... Hopefully it helps, somehow 😅

# Version and System Information
* Windows 10
```
$ node --version
v13.11.0
$ npm --version
6.13.7
```

**Comment by DanielRuf** [comment]
>  [darcyclarke](/darcyclarke) closed this [vor 1 Stunde](#event-3161218094)

@darcyclarke is this issue resolved by some commit?

**Comment by darcyclarke** [comment]
@DanielRuf apologies. This got caught up in a triage of issues in ZenHub (our project management system). I've reopened accordingly.

**Comment by gbkwiatt** [comment]
Had same issue, trying to install blank expo init project.  Node 13.12.0, npm 6.14.5
For me seems, like adding my project folder to Windows Defender exclusions helped.
`Settings->Update and Security->Windows Security->Virus & threat protection -> Virus & threat protection  settings -> Exclusions` and I have added whole folder to it.

**Comment by DanielRuf** [comment]
Did this cause the issue? I think many do not have an antivirus or Windows 10. Happens also on Linux and macOS without any realtime antivirus scanner.

In general I disable antivirus while installing to get faster installs as it scans every single file on access / creation - with disabled scripts and a full scan afterwards it works then.

**Comment by gbkwiatt** [comment]
Bit sure, but I've tried multiple times with same error, then I did that and it helped. Might be also something to do with indexing no idea.

**Comment by malisbad** [comment]
While having the `https-proxy` value set in `npm config` set, but not being on the network that had the proxy, I was getting this error. If I deleted the proxy value, things worked again. It might be a good idea to include a suggestion that you check the proxy settings for npm if people are encountering this error.

npm version: `6.17.4`
node version: `12.18.3`
nvm version: `0.35.3`

**Comment by pkrasicki** [comment]
Other related issues:
#1696
#1671
#1740
#1737
#1731
#1666
#1647
#1625
#1608
#1605
#1552
#1546
#1531
#1505
#1466
#1464
#1720
#1748

**Comment by tmtron** [comment]
We get this sometimes in Windows WSL2
* npm version: `6.14.8`
* command. `npm --unsafe-perm ci`
* but on the same system there are also some ssh/git connection-issues in WSL2 - [WSL#4690](https://github.com/microsoft/WSL/issues/4690#issuecomment-683396271), so maybe it's not an npm issue in our case

**Comment by buckleyGI** [comment]
After doing more than **500 runs** on our buildserver (Azure DevOps) we come to the the following conclusions.

- If the registry is Azure DevOps it fails in 12% of the cases with "error cb() never called!"
(There is a module called Azure artifacts which caches the npm registry)

- if the registry is npm (https://registry.npmjs.org) it succeeds **100% of the time.**

One could draw the conclusion that the fault lies with Azure DevOps but reading other reports of this issue it seems that a race condition is involved.
And 1 thing to not is that the "npm ci" command we are performing takes about 3 minutes when connecting to npm and 5 mins when connecting to DevOps.

Our setup:npm/6.14.8 node/v14.15.0 win32 x64
We tried also with the latest npm v7 which gives the same error.
Looking at the logs it happens 100% of the time (when it fails 12% of the time) after ckeditor-4.12.1 is handled.

For the moment we switched to the npm registry but we expect to be throttled at some point which was to reason to start relying on Azure Artifacts.

This issue seems elusive and quit a few people are struggling with it. **You can count on us to provide any addition information.**

**Comment by buckleyGI** [comment]
@darcyclarke Is there a way I can send the full logs (so not only the error message) in private? Thank you.

**Comment by fbeyl** [comment]
2 info using npm@6.14.5
3 info using node@v12.18.1
4 verbose npm-session ae39a2f66d8c100b
5 silly install loadCurrentTree
6 silly install readGlobalPackageData
7 http fetch GET 407 https://api.bintray.com/npm/ca/brightside/@brightside%2fcore 981ms
8 timing npm Completed in 4923ms
9 error cb() never called!
10 error This is an error with npm itself. Please report this error at:
11 error <https://npm.community>

**Comment by JohnRDOrazio** [comment]
I was just trying to setup the WordPress Gutenberg development environment, following the instructions [here](https://developer.wordpress.org/block-editor/contributors/develop/getting-started/). When running `npm ci` after several hours of downloading packages I finally got this same error:

```bash
npm WARN prepare removing existing node_modules/ before installation
npm ERR! cb() never called!

npm ERR! This is an error with npm itself. Please report this error at:
npm ERR!     <https://npm.community>

npm ERR! A complete log of this run can be found in:
npm ERR!     ~/.npm/_logs/2021-02-09T15_03_04_051Z-debug.log
```

The last few lines of the log have:

```
20451 silly extractTree glob@7.1.6 -> /mnt/e/DEVELOPMENT/WordPressDev/gutenberg/node_modules/node-gyp/node_modules/rimraf/node_modules/glob
20452 silly tarball trying glob@7.1.6 by hash: sha512-LwaxwyZ72Lk7vZINtNNrywX0ZuLyStrdDtabefZKAY5ZGJhVtgdznluResxNmPitE0SAO+O26sWTHeKSI2wMBA==
20453 silly extract semver@5.3.0 extracted to /mnt/e/DEVELOPMENT/WordPressDev/gutenberg/node_modules/node-gyp/node_modules/semver (2897ms)
20454 silly extract nopt@3.0.6 extracted to /mnt/e/DEVELOPMENT/WordPressDev/gutenberg/node_modules/nopt (2898ms)
20455 silly extract glob@7.1.6 extracted to /mnt/e/DEVELOPMENT/WordPressDev/gutenberg/node_modules/node-gyp/node_modules/rimraf/node_modules/glob (2251ms)
20456 silly extract ramda@0.25.0 extracted to /mnt/e/DEVELOPMENT/WordPressDev/gutenberg/node_modules/ramda (69428ms)
20457 timing npm Completed in 15388889ms
20458 error cb() never called!
20459 error This is an error with npm itself. Please report this error at:
20460 error <https://npm.community>
```

**NPM version**: 6.14.10
**Node version**: v14.15.4
**Ubuntu 20.04 WSL2**
______
UPDATE: I tried building the WordPress Gutenberg plugin on a different machine with the same setup, but on a different network. The packages were installed quite a bit faster and I didn't get the `cb() never called` error this time. So I'm guessing it was some kind of network issue on the machine I used the first time round.

**Comment by ChristianF88** [comment]
Hi there,

you can add **NPM version 6.14.11** to the list. **Node** has **version v15.8.0**. Everything is install stalled on:
**Linux alarmpi 5.10.13-3-ARCH #1 SMP Fri Feb 5 21:01:37 UTC 2021 armv6l GNU/Linux**

Here are the last few lines of the log file:
```
30759 silly extract caniuse-lite@1.0.30000885 extracted to /home/alarm/stackedit/node_modules>
30760 silly extract git+https://github.com/paulrosen/MIDI.js.git#abcjs extracted to /home/ala>
30761 http fetch GET 200 https://registry.npmjs.org/core-js/-/core-js-2.5.3.tgz 139130ms
30762 http fetch GET 200 https://registry.npmjs.org/caniuse-db/-/caniuse-db-1.0.30000791.tgz >
30763 silly extract caniuse-db@1.0.30000791 extracted to /home/alarm/stackedit/node_modules/.>
30764 silly extract core-js@2.5.3 extracted to /home/alarm/stackedit/node_modules/.staging/co>
30765 http fetch GET 200 https://registry.npmjs.org/fsevents/-/fsevents-1.1.3.tgz 489770ms
30766 http fetch GET 200 https://registry.npmjs.org/aws-sdk/-/aws-sdk-2.317.0.tgz 948245ms
30767 silly extract fsevents@1.1.3 extracted to /home/alarm/stackedit/node_modules/.staging/f>
30768 silly extract aws-sdk@2.317.0 extracted to /home/alarm/stackedit/node_modules/.staging/>
30769 timing npm Completed in 1277135ms
30770 error cb() never called!
30771 error This is an error with npm itself. Please report this error at:
30772 error <https://npm.community>
```

Thanks!

**Comment by GeduardoOC** [comment]
npm ERR! cb() never called!

npm ERR! This is an error with npm itself.

**Comment by mikestratton** [comment]
`npm i -g npm@latest`   

Resolved it for me.   
:-)

**Comment by BeingShaashi** [comment]
https://github.com/npm/cli/issues/417

**Comment by BeingShaashi** [comment]
The error still exists and will always I think because I see no one from NPM is actually trying to resolve it, other than telling workarounds and expecting it to work so that they can close the issue.

**Comment by hrithikwins** [comment]
0 verbose cli [
0 verbose cli   '/usr/bin/node',
0 verbose cli   '/usr/local/bin/npm',
0 verbose cli   'i',
0 verbose cli   'bootstrap@5.0.0-alpha1'
0 verbose cli ]
1 info using npm@7.6.0
2 info using node@v14.16.0
3 timing config:load:defaults Completed in 2ms
4 timing config:load:file:/usr/local/lib/node_modules/npm/npmrc Completed in 3ms
5 timing config:load:builtin Completed in 3ms
6 timing config:load:cli Completed in 3ms
7 timing config:load:env Completed in 0ms
8 timing config:load:file:/home/hrithik/workspace/react/heavenhill-2020/.npmrc Completed in 0ms
9 timing config:load:project Completed in 1ms
10 timing config:load:file:/home/hrithik/.npmrc Completed in 0ms
11 timing config:load:user Completed in 1ms
12 timing config:load:file:/etc/npmrc Completed in 0ms
13 timing config:load:global Completed in 0ms
14 timing config:load:cafile Completed in 0ms
15 timing config:load:validate Completed in 1ms
16 timing config:load:setUserAgent Completed in 1ms
17 timing config:load:setEnvs Completed in 1ms
18 timing config:load Completed in 13ms
19 verbose npm-session 4e9a670f5ee81229
20 timing npm:load Completed in 26ms
21 timing arborist:ctor Completed in 1ms
22 timing idealTree:init Completed in 2035ms
23 timing idealTree:userRequests Completed in 4ms
24 silly idealTree buildDeps
25 silly fetch manifest bootstrap@5.0.0-alpha1
26 timing arborist:ctor Completed in 0ms
27 http fetch GET 304 https://registry.npmjs.org/bootstrap 5556ms (from cache)
28 silly fetch manifest popper.js@^1.16.0
29 http fetch GET 304 https://registry.npmjs.org/popper.js 1995ms (from cache)
30 silly placeDep ROOT bootstrap@5.0.0-alpha1 REPLACE for: heavenhillresort1@0.1.0 want: 5.0.0-alpha1
31 silly placeDep ROOT popper.js@1.16.1 REPLACE for: bootstrap@5.0.0-alpha1 want: ^1.16.0
32 timing idealTree:#root Completed in 7584ms
33 silly fetch manifest popper.js@1.16.1-lts
34 silly placeDep node_modules/@material-ui/core popper.js@1.16.1-lts OK for: @material-ui/core@4.11.0 want: 1.16.1-lts
35 timing idealTree:node_modules/@material-ui/core Completed in 3ms
36 timing idealTree:node_modules/bootstrap Completed in 0ms
37 timing idealTree:node_modules/popper.js Completed in 0ms
38 timing idealTree:node_modules/@material-ui/core/node_modules/popper.js Completed in 0ms
39 timing idealTree:buildDeps Completed in 7590ms
40 timing idealTree:fixDepFlags Completed in 56ms
41 timing idealTree Completed in 9732ms
42 timing reify:loadTrees Completed in 9733ms
43 timing reify:diffTrees Completed in 64ms
44 silly reify mark retired [
44 silly reify   '/home/hrithik/workspace/react/heavenhill-2020/node_modules/bootstrap'
44 silly reify ]
45 silly reify mark retired [
45 silly reify   '/home/hrithik/workspace/react/heavenhill-2020/node_modules/popper.js'
45 silly reify ]
46 silly reify moves {
46 silly reify   '/home/hrithik/workspace/react/heavenhill-2020/node_modules/bootstrap': '/home/hrithik/workspace/react/heavenhill-2020/node_modules/.bootstrap-MMOovjIg',
46 silly reify   '/home/hrithik/workspace/react/heavenhill-2020/node_modules/popper.js': '/home/hrithik/workspace/react/heavenhill-2020/node_modules/.popper.js-fA2ADyuS'
46 silly reify }
47 timing reify:retireShallow Completed in 8ms
48 timing reify:createSparse Completed in 222ms
49 timing reify:loadBundles Completed in 0ms
50 verbose reify failed optional dependency /home/hrithik/workspace/react/heavenhill-2020/node_modules/fsevents
51 silly reify mark deleted [
51 silly reify   '/home/hrithik/workspace/react/heavenhill-2020/node_modules/fsevents'
51 silly reify ]
52 timing reifyNode:node_modules/fsevents Completed in 6ms
53 silly tarball no local data for bootstrap@https://registry.npmjs.org/bootstrap/-/bootstrap-5.0.0-alpha1.tgz. Extracting by manifest.
54 timing reifyNode:node_modules/@material-ui/core/node_modules/popper.js Completed in 512ms
55 warn deprecated popper.js@1.16.1: You can find the new Popper v2 at @popperjs/core, this package is dedicated to the legacy v1
56 timing reifyNode:node_modules/popper.js Completed in 514ms
57 http fetch POST 200 https://registry.npmjs.org/-/npm/v1/security/advisories/bulk 686ms
58 timing auditReport:getReport Completed in 694ms
59 timing auditReport:init Completed in 0ms
60 timing reify:audit Completed in 697ms
61 timing npm Completed in 428935ms
62 error cb() never called!
63 error This is an error with npm itself. Please report this error at:
64 error <https://github.com/npm/cli/issues>

**Comment by ustas4** [comment]
11869 http fetch GET 200 https://registry.npmjs.org/typescript/-/typescript-4.2.3.tgz 449483ms
11870 timing reifyNode:node_modules/typescript Completed in 469850ms
11871 warn tar TAR_ENTRY_ERROR EPERM: operation not permitted, open 'C:\Users\stas.admin\Desktop\my-app\node_modules\recursive-readdir\test\testsymlinks\linkeddir\hi.docx'
11872 silly tar Error: EPERM: operation not permitted, open 'C:\Users\stas.admin\Desktop\my-app\node_modules\recursive-readdir\test\testsymlinks\linkeddir\hi.docx'
11872 silly tar  TAR_ENTRY_ERROR EPERM: operation not permitted, open 'C:\Users\stas.admin\Desktop\my-app\node_modules\recursive-readdir\test\testsymlinks\linkeddir\hi.docx' [Error: EPERM: operation not permitted, open 'C:\Users\stas.admin\Desktop\my-app\node_modules\recursive-readdir\test\testsymlinks\linkeddir\hi.docx'] {
11872 silly tar   errno: -4048,
11872 silly tar   code: 'EPERM',
11872 silly tar   syscall: 'open',
11872 silly tar   path: 'C:\\Users\\stas.admin\\Desktop\\my-app\\node_modules\\recursive-readdir\\test\\testsymlinks\\linkeddir\\hi.docx',
11872 silly tar   entry: ReadEntry {
11872 silly tar     _events: [Object: null prototype] {},
11872 silly tar     _eventsCount: 0,
11872 silly tar     _maxListeners: undefined,
11872 silly tar     pipes: Yallist { tail: [Node], head: [Node], length: 1 },
11872 silly tar     buffer: Yallist { tail: null, head: null, length: 0 },
11872 silly tar     writable: false,
11872 silly tar     readable: false,
11872 silly tar     extended: null,
11872 silly tar     globalExtended: null,
11872 silly tar     header: Header {
11872 silly tar       cksumValid: true,
11872 silly tar       needPax: false,
11872 silly tar       nullBlock: false,
11872 silly tar       block: null,
11872 silly tar       path: 'package/test/testsymlinks/linkeddir/hi.docx',
11872 silly tar       mode: 420,
11872 silly tar       uid: null,
11872 silly tar       gid: null,
11872 silly tar       size: 0,
11872 silly tar       mtime: 2018-02-21T21:38:20.000Z,
11872 silly tar       cksum: 7331,
11872 silly tar       linkpath: '',
11872 silly tar       uname: '',
11872 silly tar       gname: '',
11872 silly tar       devmaj: 0,
11872 silly tar       devmin: 0,
11872 silly tar       atime: null,
11872 silly tar       ctime: null,
11872 silly tar       [Symbol(type)]: '0'
11872 silly tar     },
11872 silly tar     startBlockSize: 0,
11872 silly tar     blockRemain: 0,
11872 silly tar     remain: 0,
11872 silly tar     type: 'File',
11872 silly tar     meta: false,
11872 silly tar     ignore: false,
11872 silly tar     path: 'test/testsymlinks/linkeddir/hi.docx',
11872 silly tar     mode: 438,
11872 silly tar     uid: null,
11872 silly tar     gid: null,
11872 silly tar     uname: '',
11872 silly tar     gname: '',
11872 silly tar     size: 0,
11872 silly tar     mtime: 2018-02-21T21:38:20.000Z,
11872 silly tar     atime: null,
11872 silly tar     ctime: null,
11872 silly tar     linkpath: '',
11872 silly tar     absolute: 'C:\\Users\\stas.admin\\Desktop\\my-app\\node_modules\\recursive-readdir\\test\\testsymlinks\\linkeddir\\hi.docx',
11872 silly tar     [Symbol(kCapture)]: false,
11872 silly tar     [Symbol(flowing)]: true,
11872 silly tar     [Symbol(paused)]: false,
11872 silly tar     [Symbol(objectMode)]: false,
11872 silly tar     [Symbol(encoding)]: null,
11872 silly tar     [Symbol(decoder)]: null,
11872 silly tar     [Symbol(EOF)]: true,
11872 silly tar     [Symbol(emittedEnd)]: true,
11872 silly tar     [Symbol(emittingEnd)]: false,
11872 silly tar     [Symbol(closed)]: false,
11872 silly tar     [Symbol(bufferLength)]: 0,
11872 silly tar     [Symbol(destroyed)]: false
11872 silly tar   },
11872 silly tar   cwd: 'C:\\Users\\stas.admin\\Desktop\\my-app\\node_modules\\recursive-readdir',
11872 silly tar   tarCode: 'TAR_ENTRY_ERROR'
11872 silly tar }
11873 timing reifyNode:node_modules/recursive-readdir Completed in 617434ms
11874 warn tar TAR_ENTRY_ERROR EPERM: operation not permitted, open 'C:\Users\stas.admin\Desktop\my-app\node_modules\node-notifier\vendor\mac.noindex\terminal-notifier.app\Contents\Resources\en.lproj\Credits.rtf'
11875 silly tar Error: EPERM: operation not permitted, open 'C:\Users\stas.admin\Desktop\my-app\node_modules\node-notifier\vendor\mac.noindex\terminal-notifier.app\Contents\Resources\en.lproj\Credits.rtf'
11875 silly tar  TAR_ENTRY_ERROR EPERM: operation not permitted, open 'C:\Users\stas.admin\Desktop\my-app\node_modules\node-notifier\vendor\mac.noindex\terminal-notifier.app\Contents\Resources\en.lproj\Credits.rtf' [Error: EPERM: operation not permitted, open 'C:\Users\stas.admin\Desktop\my-app\node_modules\node-notifier\vendor\mac.noindex\terminal-notifier.app\Contents\Resources\en.lproj\Credits.rtf'] {
11875 silly tar   errno: -4048,
11875 silly tar   code: 'EPERM',
11875 silly tar   syscall: 'open',
11875 silly tar   path: 'C:\\Users\\stas.admin\\Desktop\\my-app\\node_modules\\node-notifier\\vendor\\mac.noindex\\terminal-notifier.app\\Contents\\Resources\\en.lproj\\Credits.rtf',
11875 silly tar   entry: ReadEntry {
11875 silly tar     _events: [Object: null prototype] {},
11875 silly tar     _eventsCount: 0,
11875 silly tar     _maxListeners: undefined,
11875 silly tar     pipes: Yallist { tail: [Node], head: [Node], length: 1 },
11875 silly tar     buffer: Yallist { tail: null, head: null, length: 0 },
11875 silly tar     writable: false,
11875 silly tar     readable: false,
11875 silly tar     extended: null,
11875 silly tar     globalExtended: null,
11875 silly tar     header: Header {
11875 silly tar       cksumValid: true,
11875 silly tar       needPax: false,
11875 silly tar       nullBlock: false,
11875 silly tar       block: null,
11875 silly tar       path: 'package/vendor/mac.noindex/terminal-notifier.app/Contents/Resources/en.lproj/Credits.rtf',
11875 silly tar       mode: 420,
11875 silly tar       uid: null,
11875 silly tar       gid: null,
11875 silly tar       size: 436,
11875 silly tar       mtime: 1985-10-26T08:15:00.000Z,
11875 silly tar       cksum: 11653,
11875 silly tar       linkpath: '',
11875 silly tar       uname: '',
11875 silly tar       gname: '',
11875 silly tar       devmaj: 0,
11875 silly tar       devmin: 0,
11875 silly tar       atime: null,
11875 silly tar       ctime: null,
11875 silly tar       [Symbol(type)]: '0'
11875 silly tar     },
11875 silly tar     startBlockSize: 512,
11875 silly tar     blockRemain: 0,
11875 silly tar     remain: 0,
11875 silly tar     type: 'File',
11875 silly tar     meta: false,
11875 silly tar     ignore: false,
11875 silly tar     path: 'vendor/mac.noindex/terminal-notifier.app/Contents/Resources/en.lproj/Credits.rtf',
11875 silly tar     mode: 438,
11875 silly tar     uid: null,
11875 silly tar     gid: null,
11875 silly tar     uname: '',
11875 silly tar     gname: '',
11875 silly tar     size: 436,
11875 silly tar     mtime: 1985-10-26T08:15:00.000Z,
11875 silly tar     atime: null,
11875 silly tar     ctime: null,
11875 silly tar     linkpath: '',
11875 silly tar     absolute: 'C:\\Users\\stas.admin\\Desktop\\my-app\\node_modules\\node-notifier\\vendor\\mac.noindex\\terminal-notifier.app\\Contents\\Resources\\en.lproj\\Credits.rtf',
11875 silly tar     [Symbol(kCapture)]: false,
11875 silly tar     [Symbol(flowing)]: false,
11875 silly tar     [Symbol(paused)]: true,
11875 silly tar     [Symbol(objectMode)]: false,
11875 silly tar     [Symbol(encoding)]: null,
11875 silly tar     [Symbol(decoder)]: null,
11875 silly tar     [Symbol(EOF)]: true,
11875 silly tar     [Symbol(emittedEnd)]: true,
11875 silly tar     [Symbol(emittingEnd)]: false,
11875 silly tar     [Symbol(closed)]: false,
11875 silly tar     [Symbol(bufferLength)]: 0,
11875 silly tar     [Symbol(destroyed)]: false
11875 silly tar   },
11875 silly tar   cwd: 'C:\\Users\\stas.admin\\Desktop\\my-app\\node_modules\\node-notifier',
11875 silly tar   tarCode: 'TAR_ENTRY_ERROR'
11875 silly tar }
11876 timing reifyNode:node_modules/node-notifier Completed in 759886ms
11877 timing npm Completed in 843699ms
11878 error cb() never called!
11879 error This is an error with npm itself. Please report this error at:
11880 error <https://github.com/npm/cli/issues>

**Comment by TaXueWuH** [comment]
npm install 

❯ npm -v
6.14.11
❯ node -v
v14.16.0

[2021-03-16T09_50_48_003Z-debug.log](https://github.com/npm/cli/files/6147840/2021-03-16T09_50_48_003Z-debug.log)

**Comment by blazardsky** [comment]
> Afaik this is resolved by force-clearing the cache and upgrading / updating to the latest npm version.

I encounterd this error while trying to _npm install responsive-loader jimp --save-dev_
Clearing the cache solved immediately this problem. Thanks @DanielRuf

**Comment by Danish-Ali097** [comment]
i encountered this error while running `npx create-react-app testapp`

**Comment by captain-aman-11** [comment]
npm ERR! cb() never called!

I encountered this error while running` meteor npm install.` Any fix for it ??

**Comment by ustas4** [comment]
thanx to @TaXueWuH it's work!

**Comment by SiddharthShyniben** [comment]
By the way, I noticed that this also happens when you abruptly terminate an installation. Just FYI.

**Comment by theBstar** [comment]
I encountered this issue while running `npx react-native init someProject`. I was on a slow network connection and the installation got terminated midday. Running `npx react-native init someProject` again resulted in `react-native command not found`. Clearing npx cache helped. https://stackoverflow.com/questions/63510325/how-can-i-clear-the-central-cache-for-npx

**Comment by SiddharthShyniben** [comment]
My probably-wrong hypothesis is that maybe when we terminate any action before the `cb()` callback gets called we have this error. As I read on the wiki, it happened because the process gets exited (by net problems, us terminating, etc.) before the callback is called. (See [this wiki](https://github.com/npm/cli/wiki/%22cb()-never-called%3F--I'm-having-the-same-problem!%22)). I'm telling you I'm probably wrong

**Comment by JayeshGohel** [comment]
Got npm ERR! cb() never called! while executing npm install. Any fix for this? Below is the image showing the warnings and errors I am getting
![npm error](https://user-images.githubusercontent.com/59264342/113042962-6ce90d80-9161-11eb-92f8-0fe2e22ab20b.PNG)

**Comment by Ally4** [comment]
Before the end of the process of npm install, I get this error
`npm ERR! cb() never called!`

**Comment by mateusgms** [comment]
I've got this error trying install dependencies  using `yarn`

**Comment by MorningK** [comment]
I got this error when i was running npm install
node -v
v14.16.0
npm -v
6.14.11
[2021-04-09T07_51_38_305Z-debug.log](https://github.com/npm/cli/files/6284052/2021-04-09T07_51_38_305Z-debug.log)

**Comment by bvaradha** [comment]
I got the same error on using this on actions on npm login. I am using mac OS as runner machine. 
npm login --scope=@<username> --registry=https://<registryURL>/npm/
  npm login --scope=@<username> --registry=https://<registryURL>/npm/
  shell: /bin/bash -e {0}
  env:
    NPM_CONFIG_USERCONFIG: /Users/XXXXXXX/actions-runner/_work/_temp/.npmrc
    NODE_AUTH_TOKEN: XXXXX-XXXXX-XXXXX-XXXXX
npm notice Log in on https://<registryURL>/npm/
npm ERR! cb() never called!

npm ERR! This is an error with npm itself. Please report this error at:
npm ERR!     <https://github.com/npm/cli/issues>
2021-04-09T11:20:51.1659890Z npm ERR! cb() never called!
2021-04-09T11:20:51.1661290Z 
2021-04-09T11:20:51.1663470Z npm ERR! This is an error with npm itself. Please report this error at:
2021-04-09T11:20:51.1665150Z npm ERR!     <https://github.com/npm/cli/issues>
2021-04-09T11:20:51.1783800Z 
2021-04-09T11:20:51.1785570Z npm ERR! A complete log of this run can be found in:

I am on macOS. 
npm -version
7.7.6
$ node --version
v15.14.0

**Comment by Prajwal-Adhav** [comment]
> ### *Updated* as of 01/15/2021
> **Note: Please read [this doc](https://github.com/npm/cli/wiki/%22cb()-never-called%3F--I'm-having-the-same-problem!%22) before filing a new issue or adding a comment to this thread.**
Hello  Everybody I am also encountering with the issue of npm ERR! cb() never called!
I will give all the possible data that I have regarding this error:-
O.S-Windows 10 Version 20H2
OS build-19042.746
npm version-v6.14.12
Node version-v14.16.1
While I was trying to download an external dependency for a React Web application I typed "npm install" in the node terminal and after some time this error popped up. This error tells that there is an error with npm itself. Below is the screenshot and log file of the same.


![Screenshot (122)](https://user-images.githubusercontent.com/73768647/114262967-6f205700-9a00-11eb-90e4-2b91b22825a9.png)

**[2021-04-10T07_21_59_124Z-debug.log](https://github.com/npm/cli/files/6289448/2021-04-10T07_21_59_124Z-debug.log)**

**Comment by 0xPolybit** [comment]
![npm-not-working](https://user-images.githubusercontent.com/62232457/114379422-6a95a300-9ba6-11eb-8a5e-60b5bb9762e8.PNG)
I was trying to create a React app for my personal project, but then this error pops up whenever I'm in the middle of downloading `react-dom`. I see several people facing this issue, so I think this is a problem with npm itself.
Please help us.

**Comment by 0xPolybit** [comment]
I found a quick fix. Don't know why it worked, but if you're on **Windows**, fixing the problem is as simple as going to the **Control Panel** > **Uninstall a program**, then selecting **Node.js** and clicking **Repair**. It works!

**Comment by sunshu** [comment]
Update npm to the latest version fixed my question

**Comment by woody-li** [comment]
Same error, already repair Node.js.
```
0 verbose cli [
0 verbose cli   'C:\\Program Files\\nodejs\\node.exe',
0 verbose cli   'C:\\Users\\Administrator\\AppData\\Roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js',
0 verbose cli   'i'
0 verbose cli ]
1 info using npm@7.9.0
2 info using node@v14.16.1
3 timing npm:load:whichnode Completed in 0ms
4 timing config:load:defaults Completed in 0ms
5 timing config:load:file:C:\Users\Administrator\AppData\Roaming\npm\node_modules\npm\npmrc Completed in 0ms
6 timing config:load:builtin Completed in 0ms
7 timing config:load:cli Completed in 16ms
8 timing config:load:env Completed in 0ms
9 timing config:load:file:C:\proj\.npmrc Completed in 0ms
10 timing config:load:project Completed in 0ms
11 timing config:load:file:C:\Users\Administrator\.npmrc Completed in 15ms
12 timing config:load:user Completed in 15ms
13 timing config:load:file:C:\Users\Administrator\AppData\Roaming\npm\etc\npmrc Completed in 0ms
14 timing config:load:global Completed in 0ms
15 timing config:load:validate Completed in 0ms
16 timing config:load:setEnvs Completed in 0ms
17 timing config:load Completed in 47ms
18 timing npm:load:configload Completed in 47ms
19 timing npm:load:setTitle Completed in 0ms
20 timing npm:load:setupLog Completed in 0ms
21 timing npm:load:cleanupLog Completed in 0ms
22 timing npm:load:configScope Completed in 0ms
23 timing npm:load:projectScope Completed in 0ms
24 timing npm:load Completed in 63ms
25 timing config:load:flatten Completed in 16ms
26 timing arborist:ctor Completed in 15ms
27 timing arborist:ctor Completed in 0ms
28 timing idealTree:init Completed in 93ms
29 timing idealTree:userRequests Completed in 0ms
30 silly idealTree buildDeps
31 silly fetch manifest @babel/core@^7.13.10
32 http fetch GET 200 https://npm.org/@babel%2fcore 5655ms
//Hide some 200 package request.
45 silly fetch manifest babel-loader@^8.2.2
46 http fetch GET 200 https://npm.org/babel-loader 12799ms
47 silly fetch manifest webpack@^5.28.0
48 timing npm Completed in 155688ms
49 error cb() never called!
50 error This is an error with npm itself. Please report this error at:
51 error <https://github.com/npm/cli/issues>
```

**Comment by keshaviagg** [comment]
> I think we have many reports of the same so these should all be duplicates.
> 
> #423
> #425
> #442
> #451
> #455
> #465
> 
> Afaik this is resolved by force-clearing the cache and upgrading / updating to the latest npm version.

its not working still facing the same issue

**Comment by cramirez201** [comment]
cb() never called! During installation of new Angular project...

node -v:
v14.16.1

npm -v:
7.10.0

SO: W10 Pro x64

Version: 20H2

![image](https://user-images.githubusercontent.com/82670652/115057862-8fde2600-9eb2-11eb-8a08-e02b36325d78.png)

Log: 
[2021-04-16T16_50_36_767Z-debug.log](https://github.com/npm/cli/files/6326750/2021-04-16T16_50_36_767Z-debug.log)

Help pls!

**Comment by Karmavil** [comment]
Hi. thanks for this note it was helpful:

> ### *Updated* as of 01/15/2021
> 
> **Note: Please read [this doc](https://github.com/npm/cli/wiki/%22cb()-never-called%3F--I'm-having-the-same-problem!%22) before filing a new issue or adding a comment to this thread.**

I'm not going to give you too many details basically because I will probably give you wrong or not related information but I solved this issue that **in my case** it was related with this specific package as a dependency: amplify-cli (npm intall --save amplify-cli)

This is what happened 
* I removed the package.
* I upgrade npm
* I tried to install amplify-cli with save again then I got the _ERR! cb()_ error

Details:
* I had installed this package before (as I said)
* I had installed this dependencies: <ins>graphql</ins> and <ins>@types/grapthl</ins> (I cannot tell if it was a mistake install those)
* I remove them, then ran npm install --save amplify-cli and the install worked fine (this package was the one giving me the _ERR! cb()_ error message) 

I understand this errors are specific, but maybe it could be helpful if this thing about @types/ is the problem

## Important

Now I'm in a nightmare called  **Try `npm install @types/graphql` if it exists** so take you time and think twice if you're gonna do that 🙅🏽  I cannot remember how did I solve this before

#### Solved

Jesus! that was horrible.. The only way I could solve that **try to intall** bulls~~ was restarting the container after `npm install --save` but I did it before without restart and it did not work.. I cannot imagine if it is a problem with VScode, angular, npm, cache.. you name it. Luck for me it was a container

**Comment by quaccer** [comment]
I got same error

**Comment by hamzakilic** [comment]
I got same error. and **FIXED IT**. 
I was trying to dockerize a nodejs project with a docker file. and package.json.
on my ubuntu machine there is no problem, but on CI/CD machine (centos,debian) almost fails to dockerize

**Tried Steps:**

-  clear node_modules
-  updating node version
- updating npm version
- centos 
- debian
- docker with different versions
-  lots of configuration (npm install fetch-retries, timeouts ....)


**My problem:**

My problem was at **TCP level.** 
some of the **TCP** connections to **registry.npmjs.org** was returning **RESET**.
with tcpdump I saw that if **npm install fails cb() never called,** I got TCP RESET packets from registry.npmjs.org ips.
if npm install success, then no TCP RESET from registry.npmjs.org ips. 
I dont know why TCP RESET occurs. big network problem

**My Solution:**
I putted  **registry.npmjs.org** behind of an **NGINX proxy** on K8
and changed dockerfile 
**npm install** ====> **npm install --strict-ssl=false** (because nginx will return a default ssl)

**docker build -t tag myproject --add-host registry.npmjs.org:nginxip .**  (redirect all npm install trafic to your nginx)

below is the sample K8 ingress

```yaml
apiVersion: v1
kind: Namespace
metadata:
  name: gitlab
  labels:
    app.kubernetes.io/name: gitlab
    app.kubernetes.io/part-of: gitlab
---
apiVersion: v1
kind: Service
metadata:
  name: npm-service
  namespace: gitlab
spec:
  externalName: registry.npmjs.org
  type: ExternalName
---
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: ingress-npm-gitlab
  namespace: gitlab
  annotations:
    kubernetes.io/ingress.class: "nginx"
    nginx.ingress.kubernetes.io/ssl-redirect: "true"
    nginx.ingress.kubernetes.io/force-ssl-redirect: "true"
    nginx.ingress.kubernetes.io/upstream-vhost: "registry.npmjs.org"
    nginx.ingress.kubernetes.io/ssl-passthrough: "false"
    nginx.ingress.kubernetes.io/backend-protocol: "HTTPS"
spec:
  rules:
    - host: registry.npmjs.org
      http:
        paths:
          - path: /
            pathType: Prefix
            backend:
              service:
                name: npm-service
                port:
                  number: 443`

`

**Comment by KogaiIrina** [comment]
Hi! I got this error when I was running npm install
node -v
v14.16.0
npm -v
7.10.0

____________________________

I solved this problem by updating my npm

**Comment by HomyeeKing** [comment]
the only suck thing I know is that  the warning message of `npm publish`   is indirect and  not clear
 when you have not verify your email, just tell that they haven't verified the email, not something like security policy thing and package version problem...

**Comment by KZRepository** [comment]
>npm i
npm ERR! cb() never called!

npm ERR! This is an error with npm itself. Please report this error at:
npm ERR!     <https://github.com/npm/cli/issues>

fetch GET 304 https://registry.npmjs.org/grunt-cssnano 784ms (from cache)
1900 timing metavuln:packument:grunt-cssnano Completed in 786ms
1901 timing metavuln:load:security-advisory:grunt-cssnano:EtYBVUQGQLlhlpJr8KS/6Ije2vBT1OrDb4ZylGjBbqEpHu5oNhsP45j3VEbnmDsxSDP8IWOiIk4/CNRyBk/RpQ== Completed in 1ms
1902 timing metavuln:cache:put:security-advisory:grunt-cssnano:6zfI7LzZ7jhTaIHReTadx2z1+1e771qctuX2ezhF2HjxeHbwezKe6xIw4RwD9LlZN7IMyqdkoz3KNhgkx78nwQ== Completed in 10ms
1903 timing metavuln:calculate:security-advisory:grunt-cssnano:EtYBVUQGQLlhlpJr8KS/6Ije2vBT1OrDb4ZylGjBbqEpHu5oNhsP45j3VEbnmDsxSDP8IWOiIk4/CNRyBk/RpQ== Completed in 797ms
1904 timing auditReport:init Completed in 24937ms
1905 timing reify:audit Completed in 27613ms
1906 http fetch GET 200 https://registry.npmjs.org/caniuse-lite/-/caniuse-lite-1.0.30001228.tgz 25743ms
1907 timing reifyNode:node_modules/caniuse-lite Completed in 27883ms
1908 timing npm Completed in 43488ms
1909 error cb() never called!
1910 error This is an error with npm itself. Please report this error at:
1911 error <https://github.com/npm/cli/issues>

**Comment by Ubaid-Ali** [comment]
Worked for me but I don't know is it true or not,

I was getting the same error when I was trying to install "netlify-cli" command is C:Users>MyName>npm install netlify-cli -g,
I upgraded npm version LTS to LATEST with this command >npm i -g npm@latest,
Still not work then,
I just **deleted the log file** from here ~/AppData/Local/npm-cache/_logs/2021-05-19T18_46_01_599Z-debug.log **using git bash**,
$ rm ~/AppData/Local/npm-cache/_logs/2021-05-19T18_46_01_599Z-debug.log,
which was showing with the error in cli,
Try again, Installed Successfully.

**Comment by RedHoodJT1988** [comment]
Hello, I ma receiving this error while using WSL2 Ubuntu 20.04. I have installed node and npm via nvm. I am trying to run npx create-react-app projectname but I keep running into the issue. I am not sure what is causing it as I have the latest version of both npm and npx. Any assistance on this would be very helpful. Thanks.

**Comment by darcyclarke** [comment]
npm `v6` is no longer in active development; We will continue to push security releases to `v6` at our team's discretion as-per our [Support Policy](https://github.com/npm/cli/wiki/Support-Policy).

If your bug is preproducible on `v7`, please re-file this issue using our [new issue template](https://github.com/npm/cli/issues/new?assignees=&labels=Bug%2CNeeds+Triage%2CRelease+7.x&template=bug_7.yml&title=%5BBUG%5D+%3Ctitle%3E).

If your issue was a feature request, please consider opening a new [RRFC](https://github.com/npm/rfcs/issues) or [RFC](https://github.com/npm/rfcs). If your issue was a question or other idea that was not CLI-specific, consider opening a discussion on our [feedback repo](https://github.com/npm/feedback)

**Closing:** This is an automated message.

**Comment by darcyclarke** [comment]
npm `v6` is no longer in active development; We will continue to push security releases to `v6` at our team's discretion as-per our [Support Policy](https://github.com/npm/cli/wiki/Support-Policy).

If your bug is preproducible on `v7`, please re-file this issue using our [new issue template](https://github.com/npm/cli/issues/new?assignees=&labels=Bug%2CNeeds+Triage%2CRelease+7.x&template=bug_7.yml&title=%5BBUG%5D+%3Ctitle%3E).

If your issue was a feature request, please consider opening a new [RRFC](https://github.com/npm/rfcs/issues) or [RFC](https://github.com/npm/rfcs). If your issue was a question or other idea that was not CLI-specific, consider opening a discussion on our [feedback repo](https://github.com/npm/feedback)

**Closing:** This is an automated message.
